addons/mail/static/src/models/rtc/rtc.js

```diff
@@ -74,9 +74,19 @@ function factory(dependencies) {
             this._onKeyUp = this._onKeyUp.bind(this);
             browser.addEventListener('keydown', this._onKeyDown);
             browser.addEventListener('keyup', this._onKeyUp);
-            browser.addEventListener('beforeunload', async () => {
-                this.channel && await this.channel.leaveCall();
-            });
+            /**
+             * Disconnects the RTC session if the page is closed or reloaded,
+             * it is done in both 'unload' and 'beforeunload' as browsers may
+             * trigger only one of these events in different cases.
+             */
+            const onAnyUnload = async () => {
+                this.channel && await await this.async(() => this.env.services.rpc({
+                    route: '/mail/rtc/channel/leave_call',
+                    params: { channel_id: this.channel.id },
+                }, { shadow: true }));
+            };
+            browser.addEventListener('beforeunload', onAnyUnload);
+            browser.addEventListener('unload', onAnyUnload);
```

Comment:
Move as an instance method instead of having it in scope, remove the listered in will delete (to be able to test it later).